caluma/caluma_form/tests/test_form.py

```diff
@@ -106,28 +107,36 @@ def test_save_form_created_as_admin_user(db, form, admin_schema_executor, admin_
     assert result.data["saveForm"]["form"]["createdByUser"] == admin_user.username
 
 
+@pytest.mark.parametrize("use_graphql", [True, False])
 @pytest.mark.parametrize("form__meta", [{"meta": "set"}])
-def test_copy_form(db, form, form_question_factory, schema_executor):
+def test_copy_form(db, form, form_question_factory, schema_executor, use_graphql):
```

Comment:
The `use_grapql` parameter splits the test in two parts, with almost no common code. Not worth the parametrisation. Just write two tests, it will be much more readable.

Reply:
Actually, this is exactly how we did it for the most of the other extracted API functions. This test shares like 5 assertions and _most importantly_ the same input to make sure the both APIs do the same.

Reply:
The places where we did it are getting uglier each time we need to extend them. It was a bad idea from the start, just not obvious back then. And we can still factor out the asserting code. Still cleaner IMHO

Reply:
Ok, we now have a duplicated test.